lib/mappings/createNew.js

```diff
@@ -1,7 +1,38 @@
-import { createWallet } from '../actions';
+import assert from 'bsert';
+import { selectWallets, selectCurrentChain, selectNetwork } from './selectors';
+import { createWallet, createAccount } from '../actions';
+import { getxpub } from '../utilities';
+
+/*
+ * getxpub then create account using the xpub
+ */
+
+async function createHardwareWatchOnlyAccount(walletId, accountId, options) {
+  return async function action(dispatch) {
+    const { xpub } = await getxpub({
+      chain: options.chain,
+      network: options.network,
+      account: options.account,
+      hardwareType: options.hardwareType,
+    });
+
+    assert(walletId, 'must pass wallet id');
+    assert(accountId, 'must pass account id');
+
+    dispatch(
+      createAccount(walletId, accountId, {
+        watchOnly: true,
+        xpub,
+      })
+    );
+  };
+}
 
 function mapStateToProps(state, otherProps) {
```

Comment:
I'm not sure if I understand your comment. Each `mappings/*` corresponds to a `containers/*`.

This was done to modularize the `imports` and make the `container` files cleaner

Reply:
Ah, ok. Sorry about that. I misread this as the bPanel level extension API (e.g. [here](https://github.com/bpanel-org/mempool-widget/blob/master/lib/index.js#L31)).

For what it's worth, the convention I've normally seen in React + Redux is to modularize by making a container for these "mappings", i.e. containers have state and/or connect to redux store, and components are wrapped in the containers. There's a stackoverflow thread linked to from redux docs [here](https://stackoverflow.com/questions/32634320/how-to-structure-redux-components-containers/32921576), and the bpanel components [use this as well](https://github.com/bpanel-org/bpanel/tree/development/webapp/containers). This could work too but I think this is probably why I got confused.